src/resources/date-utilities.ts

```diff
@@ -0,0 +1,353 @@
+import { isDate } from 'lodash'
+import { ApiDateTimeUtc, IsoDatePacific } from '@bcrs-shared-components/interfaces'
+
+const MS_IN_A_DAY = (1000 * 60 * 60 * 24)
+
+export default class DateUtilities {
+  /**
+   * Fetches and returns the web server's current date (in UTC).
+   * Used to bypass the user's local clock/timezone.
+   * Ref: https://www.npmjs.com/package/serverdate
+   * @returns a promise to return a Date object
+   */
+  static async getServerDate (): Promise<Date> {
```

Comment:
FYI, I want to eventually replace this with an external library, or our web worker, or whatever, to:
- reduce the number of network calls
- simplify the app code

All the while having "date and time synced to a known good reference" available in this UI. The reason is that we can't (shouldn't) trust the browser's date-time. For some things, we need time accuracy down to about a minute.

Reply:
I'm not sure I understand the reason for getting the dateTime from the server.  If there's a filing that must be submitted before midnight.  We make the deadline clear on the form and then, on submission, validate the submission time as part of the business logic.  If the form is submitted past the deadline, we reject the submission and return a specific error message from the API.  I think I'm missing some aspect of this.  Does the complexity lie when handling submissions from parts of the province on mountain time rather than pacific?

Reply:
First, all times are Pacific (Daylight or Standard)

Some examples:
- appointment / cease date for directors
- future effective date (and time)

The above need to know accurate "now" time and date.